### ISSUE TITLE
Firefox support

### ISSUE THREAD

**Issue by seb-koemei** [ZenHubIO/support#30]
Adding Firefox support would be nice.

**Comment by jalberto** [comment]
+1

**Comment by ametaireau** [comment]
Hi, do you have any information about when this will be available? Even a first version, released as alpha software could help you to get to more users here.

Also, if you have performances problems, can you consider being more explicit about them so we can help you to get past them?

Thanks!

**Comment by TheBrockEllis** [comment]
Any update on the progress? Zenhub looks super promising for our team of 10 but many of us value our privacy and aren't interested in switching away from Firefox. Any chance we could open a bounty on this? I'd be willing to fork over some cash to get some Firefox love.

**Comment by weirdguy** [comment]
+1

**Comment by jwir3** [comment]
I might be able to help with a firefox extension if you need help - I've written a number of them, and I used to work for mozilla as a layout dev, so I'm pretty familiar with the API.

**Comment by caffeinewriter** [comment]
Still excited to see this. Hoping to hear an ETA soon.

**Comment by sandipp** [comment]
+1 for FF support

**Comment by chishaku** [comment]
+1

**Comment by azenMatt** [comment]
Hi all, thank you for the support and offers to help with our Firefox extension. I wanted to let you all know that it has not fallen off our radar. At the moment we have been focused on making sure our Chrome extension is as fast and reliable as possible before we start supporting multiple browsers. 

We are currently really happy with the Chrome extension and will have more information on the Firefox version for everyone soon. Thank you for your patience!

**Comment by gdude2002** [comment]
Would it perhaps be better to lock this ticket now that it's confirmed to be in the pipeline?

**Comment by irvincee** [comment]
My team and I are very excited to incorporate Zenhub into our workflow! I see Zenhub as a great alternative to Pivotal. Our Android developer uses Firefox and doesn't want to be left out of the action! Hope there are developments on the Firefox extension soon.

**Comment by cander21** [comment]
:+1:

**Comment by derrickj3000** [comment]
:+1:

**Comment by jeljaik** [comment]
:+1:

**Comment by gfody** [comment]
:+1:

**Comment by hamen** [comment]
:+1:

**Comment by kud** [comment]
By the way, why some browsers aren't supported? You use a specific feature of chrome?

**Comment by gdude2002** [comment]
@kud Probably because extensions are all done differently for different browsers, I assume

**Comment by kud** [comment]
Now, it's mostly JavaScript. I've already done some chrome and firefox
addons.

**Comment by elliot-borst-yuranga** [comment]
This issue was raised in November of 2013 and its now April 2015, When is support for other browsers coming?

**Comment by unixfox** [comment]
:+1:

**Comment by emanuelecasadio** [comment]
+1

**Comment by wvengen** [comment]
:+1:

**Comment by shouze** [comment]
:+1:

**Comment by hellsan631** [comment]
+1

**Comment by marcbertone** [comment]
:+1:

**Comment by TzvetomirTodorov** [comment]
:1:

**Comment by abacaj** [comment]
:+1: wtf.

**Comment by caffeinewriter** [comment]
@azenMatt Any word on FF support?

**Comment by azenMatt** [comment]
@caffeinewriter - this is still a high priority for us, no exact ETA yet but it is a really important part of out roadmap.

**Comment by metamatik** [comment]
:+1:

**Comment by kud** [comment]
`seb-koemei opened this Issue on 28 Nov 2013`

**Comment by tshirtman** [comment]
:+1:

**Comment by dessant** [comment]
![thumbs-up_group](https://cloud.githubusercontent.com/assets/7513068/8272277/0d10e798-1845-11e5-8826-2b747e3bade3.jpg)

**Comment by matham** [comment]
:+1:

**Comment by kud** [comment]
![](http://i.giphy.com/lFlZd7ZmoUqVa.gif)

**Comment by elliot-borst-yuranga** [comment]
:+1: 

:+1:


:+1:


:+1:


:+1:


:+1:

**Comment by ahmed-sghaier** [comment]
:+1:

**Comment by Baires** [comment]
:+1:

**Comment by andreackinney** [comment]
+1,000,000

**Comment by onlimii** [comment]
:+1:

**Comment by sou-lab** [comment]
:+1:

**Comment by jtroe** [comment]
I'm with @andreackinney (:+1: * 1000000.0)

**Comment by neiker** [comment]
:+1:

**Comment by ldonnet** [comment]
+1 with firefox

**Comment by fcarrega** [comment]
:+1:

**Comment by jemc** [comment]
:+1:

**Comment by brechtvdv** [comment]
:+1:

**Comment by bleda** [comment]
:+1:

**Comment by eistrati** [comment]
+1 please add firefox support

**Comment by jrabolli** [comment]
friends don't let friends use chrome

**Comment by kunallillaney** [comment]
:+1:

**Comment by FilWisher** [comment]
:+1:

**Comment by trodrigu** [comment]
(Y)

**Comment by nniiicc** [comment]
+1

**Comment by ginfuru** [comment]
:+1:

**Comment by Cyral** [comment]
:+1:

**Comment by ametaireau** [comment]
This issue was raised in November of 2013 and one of the first comments was stating that there is an early version of a Firefox extension.

 I don't think there is much coming down trough this pipe, unfortunately.

**Comment by tzaeschke** [comment]
:+1:

**Comment by KevinGrandon** [comment]
I actually stopped using ZenHub because of this issue. Will consider going back in the future if it's resolved.

**Comment by roham** [comment]
> This issue was raised in November of 2013 and one of the first comments was stating that there is an early version of a Firefox extension.

There was! But as we added more functionality to the Chrome version, Firefox got increasingly out of date :) 

We hear you loud and clear, and would love to make you happy. That said, our priority is still accelerating functionality for our existing users because the vast majority of folks in our industry use Chrome and are quite happy with it. As soon as we knock out a few remaining important improvements to the core product, we will tackle ZH outside of Chrome.

**Comment by kud** [comment]
Chrome is the new IE.

**Comment by asghaier** [comment]
@roham I also gave up on ZenHub too, since it is a real pain for me to start a dedicated chrome instance for just this feature.
1/3 of my team is using Firefox and will not give up on this browser for a single feature. Hopefully you will be able to make it on time (which doesn't seem to have a slot/date yet on your roadmap).

**Comment by kud** [comment]
I understand your point of view ZenHub but as I work in the web, it's just unacceptable for me to read that you don't care about supporting major current browsers.

**Comment by roham** [comment]
> I understand your point of view ZenHub but as I work in the web, it's just unacceptable for me to read that you don't care about supporting major current browsers.

We absolutely care, don't get me wrong. When I say "can't give a date yet, but soon" -  I mean it. Non-chrome use of ZenHub is one of the highest priority items on our roadmap today – I was explaining why it's been so delayed. As I indicated, we're close to being able to start knocking this out of the park for you guys.

**Comment by kud** [comment]
Okay it's more clear to me. Thank you. ;)

**Comment by gbonfant** [comment]
:+1:

**Comment by kalmbach** [comment]
:+1: Firefox support please :)

**Comment by mattduck** [comment]
:+1:

**Comment by charliewolf** [comment]
+1

**Comment by matham** [comment]
https://wiki.mozilla.org/WebExtensions should make it easier.

**Comment by futuresoup** [comment]
:+1: pretty please :)

**Comment by webaaz** [comment]
:+1:

**Comment by masseuro** [comment]
:+1:

**Comment by rafspiny** [comment]
Looking forward to use the FF version

**Comment by rnaimi** [comment]
+1

**Comment by moylop260** [comment]
:+1:

**Comment by JacopoGobbi** [comment]
:+1:

**Comment by caffeinewriter** [comment]
I'm just going to chime in once more. I understand that your Chrome product is probably your main focus, but the fact that it's taken so long for any word on the Firefox version, it's really offputting. While I'm sure you're working hard on it behind the scenes, the fact that your Chrome extension is receiving all the love, leaving the Firefox version even further in the dust makes me feel like other browsers will always be second-class citizens in terms of support. It's entirely possible that this isn't the case, but the delay's been long enough that it's very unlikely that I'll be using Zenhub, even when Firefox support is available. I mean, it's been nearly two years since the issue was opened, and almost a year since we were told to "watch this space". I don't want to discourage you guys, since this would be amazing feature, but I do want to emphasize that the fact that the promise of "it's coming" has started to become stale, and has left a sour taste in my mouth overall. Good luck ZenHub.

**Comment by kud** [comment]
Same here. Web is not only one browser.

**Comment by ninjatux** [comment]
:+1:

**Comment by charliewolf** [comment]
:+1:

**Comment by jaschweder** [comment]
:+1:

**Comment by andreimamlay** [comment]
:+1:

**Comment by azenMatt** [comment]
Hi all, 

A quick update for everyone one this, 

@caffeinewriter - to answer:

> the fact that your Chrome extension is receiving all the love, leaving the Firefox version even further in the dust makes me feel like other browsers will always be second-class citizens in terms of support.

I want to be clear that the opposite is true :smile: the reason this has taken us longer than expected is because we want to ensure that both extensions are treated the same - which means we need as much shared code as possible so that we can always keep them in feature parity. 

The good news is, that with the recent firefox updates, this has become much easier - and we have made some great progress on refactoring our extension to work on both browsers. 

Pictures say more than words, so I think it is easiest for me to share this:

![image](https://cloud.githubusercontent.com/assets/7584089/10860019/62a26654-7f20-11e5-8431-b7f7f99c1589.png)

We are racing to get this to you as soon as possible - we have an internal target of being live before Thanksgiving - watch this space!

**Comment by jaypozo** [comment]
:+1:

**Comment by robeastwood** [comment]
:+1: Waiting for this feature before I sign up

**Comment by cog-ethz-git** [comment]
How soon after thanksgiving? I am currently setting up our project environment and I would love to go for zenhub, but if it takes to long I will have to choose an alternative.

**Comment by azenMatt** [comment]
@cog-ethz-git - it will be ready in time for your new project :) updates coming before all the Turkey leftovers are gone!

**Comment by jhnns** [comment]
> :+1: Waiting for this feature before I sign up

Same here!

**Comment by acidicX** [comment]
can't wait! :+1:

**Comment by mcobzarenco** [comment]
+1

**Comment by nadirayasmeen** [comment]
:+1:

**Comment by carloshfoliveira** [comment]
So, is it ready for Firefox/Iceweasel?

**Comment by vicentebolea** [comment]
Same request!

**Comment by loint** [comment]
:+1:

**Comment by cog-ethz-git** [comment]
Could you please [sign](https://developer.mozilla.org/en-US/Add-ons/Distribution) the firefox addon?

**Comment by cog-ethz-git** [comment]
I just installed the firefox extension on the developer edition, besides the previously mentioned fact that your extension is not signed it just crashes my browser completely. As soon as I visit a git page all crashes and I can't do anything. I disabled the extension for now.

I hope you get a hang on this, really soon.

**Comment by azenMatt** [comment]
It’s here! 

![image](https://cloud.githubusercontent.com/assets/7584089/11538969/7626e114-98d9-11e5-9475-9b29c0ed4425.png)

ZenHub now officially has Firefox support :)

You can download it from the main website, or we threw up a landing page here - https://www.zenhub.io/firefox - that also contains some details for a discount and some swag to celebrate it’s launch! 

ZenHub for firefox has also made it onto https://www.producthunt.com/ today if you want to jump into the conversation there. 

Enjoy - and let me know your feedback! We have worked hard to make sure the experience is exactly the same as the Chrome experience - so that as we release future features and patches - both versions stay in sync. 

-------------

(also, the fact that the issue page side bar scrolls down the page with you in firefox is amazing! GitHub need to include that in the chrome version or I am sticking with FF as my main browser :wink:  )

**Comment by azenMatt** [comment]
@cog-ethz-git 

For the signing issue, we decided that we would launch without this as we kept having to push back whilst we waited for the FF team. As it would happen, the signed version came through this morning! so we will get that up right away! 

We will be staying with a self hosted (but signed) extension for the moment through, as downloads through the FF market place currently have a 7 week review queue. As we do not want the Chrome and FF extensions to get out of sync, a 7 week delay is not going to be workable! 

In terms of your issues, I haven’t been able to replicate, would you be able to send me an email at support@zenhub.io with some more details and we will get this working for you!

**Comment by jhnns** [comment]
Nice :+1:

**Comment by cog-ethz-git** [comment]
@azenMatt 
It seems to work on another machine I work with. I will revisit the non-working machine tomorrow and try to get you an error log of what happened.

**Comment by nadirayasmeen** [comment]
Thank you!!  :+1:

**Comment by carloshfoliveira** [comment]
Now I can get rid of Chrome :+1: 
![ZenHub on FF](https://files.zenhub.io/565f4d09bd1f5c922032fed9)
Uploaded using [**ZenHub.io**](http://www.zenhub.io)

**Comment by carloshfoliveira** [comment]
For those who are facing problems with the unsigned package(it happens mostly with dev edition/nightly), there's a workaround, though it's up to you to do it.

**Comment by geckotang** [comment]
Congratulations! :+1:

**Comment by rafspiny** [comment]
Congratulations! This is great.

Thank you.

2015-12-03 0:43 GMT+00:00 Shotaro Sakamaki <notifications@github.com>:

> Congratulations! [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ZenHubIO/support/issues/30#issuecomment-161479310>.
>



-- 
Raffaele Spinelli
Ph.D. Student
University of Salerno
_______________________________________
e-mail: spinelli <at> dia.unisa.it
skype :  Raffaele Spinelli

**Comment by Mathieuu** [comment]
The Firefox extension is now signed.

**Comment by carloshfoliveira** [comment]
Great! No more need for the workaround :D

**Comment by TPS** [comment]
Hey, @ZenHubIO, can you all get this (& all previous versions) posted on [AMO](https://addons.mozilla.org/) & [APO](https://addons.palemoon.org/)?  Ff, forks, & related software (e.g., [Pale](https://forum.palemoon.org/viewtopic.php?t=7426) [Moon](https://forum.palemoon.org/viewtopic.php?t=7426)) ___really___ don't like self-hosted XPIs (for security reasons), & having it posted there solves that.  Also, it makes browser requirements more transparent, which is helpful for such.

**Comment by azenMatt** [comment]
Hi @TPS - we hope to be able to do this soon, 

The firefox review process is getting much better - as of the latest updates from the firefox team 95% of updates are being reviewed in less than 5 days. 

Compared with June 2015 when it was 10 weeks for a review 😬 


Due to our need to be reactive to GitHub changes, 5 days is still too long for us - but as this comes closer inline with the Chrome review policy we will be able to move to this method. 

I would much rather this goes through AMO than our current process, so when it makes sense we will switch over
